wordpress-mobile/WordPress-iOS
# [CLOSED] Jetpack Social: Update external services endpoint to v2

Resolves #20676
Depends on https://github.com/wordpress-mobile/WordPressKit-iOS/pull/600 👈🏼 Please review this first.

- The `PublicizeService` model is now updated to have a `status` field with a default value of `"ok"`.
- Switched the remote call in `syncPublicizeServicesForBlog` to use the v2 version.

## To test

### Verifying Core Data model migration

- Install a `trunk` version of the Jetpack app (or any version with model version 148). 
- Launch the app, and log in with your WordPress.com account.
- Go to Site Menu > Social to sync PublicizeService data.
- Overwrite the app with the version of this branch.
- Verify that the app does not crash.

### Verifying the status value

Since there's no UI changes, the easiest way would be to put a breakpoint in this line:

https://github.com/wordpress-mobile/WordPress-iOS/blob/0c1e5381ffd5395b7af1db8360495c247ad8ada0/WordPress/Classes/Services/SharingService.swift#L352

and print out the assigned `status` value in the console:

```
po pubService?.status
```

Here are the test steps:

- Prepare a test site with the conditions outlined in pctCYC-OI-p2#detecting-when-twitter-is-stopped.
- Launch the app, and log in with your WordPress.com account.
- Switch to the test site.
- Go to Site menu > Social. The breakpoint should be triggered here.
- 🔎 Verify that the value is `unsupported` for the disabled service, and `ok` for others.

## Regression Notes
1. Potential unintended areas of impact
This PR adds a new field for `PublicizeService` model. This is currently only used within the Sharing menu.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manually tested the changes, and added some unit tests in the dependency library.

3. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:


UI Changes testing checklist:

N/A. There are no visual changes.
